screens/BooksScreen.js

```diff
@@ -0,0 +1,69 @@
+import React, { useEffect, useState } from 'react';
+import { ActivityIndicator, FlatList, StyleSheet, Text, View } from 'react-native';
+
+export default function BooksScreen() {
+  const [isLoading, setLoading] = useState(true);
+  const [data, setData] = useState([]);
+
+  //async function is returning a promise , sempre
+  const getBooks = async () => {
+    try {
+      const response = await fetch('https://openlibrary.org/subjects/painting.json');
+      const json = await response.json();
+      setData(json.works);
+    } catch (error) {
+      console.error(error);
+    } finally {
+      setLoading(false);
+    }
+  }
+
+  useEffect(() => {
+    getBooks();
+  }, []);
+
+
+  const renderItem = ({ item }) => {
+
+    const authors = item.authors.length > 1
+      ? `[authors: ${item.authors[0].name} and ${item.authors.length - 1} more]`
+      : `[author: ${item.authors[0].name}]`;
+
+    const isbnArray = typeof item.availability === "object" ? item.availability : '' ;
+    const isbn = typeof isbnArray.isbn === "string" ? `[ISBN: ${isbnArray.isbn}]` : null ;
```

Comment:
this is a very roundabout way of doing it, but it works.

however it is confusing to read and should be changed.

what i dislike about this solution:
* `item.availability` can be either `undefined` or an `object`.
* if it is an `object`, you keep it, if it is `undefined` you turn it into a `string`.
* then you check if an entry of the `isbnArray` (which is now either a `string` or an `object`, is a `string`
  * it can be easily misread when taking a quick glance, that you check if `isbnArray` itself is a string, as that is one of the cases you specifically introduced in the previous line
  * the likelihood of misreading is further increased by the fact that the `isbn` key does not exist on a `string` and therefore checking the type of it is not necessarily intuitive
* also: what happens if `item.availability` is `null` instead of `undefined`? (currently that does not seem to occur)

Reply:
This happens if `item.availability` is `null` instead of `string` 
![IMG_7AF730E53B4C-1](https://user-images.githubusercontent.com/43641321/141811888-172f585f-14f0-48ae-aadb-1d91ed35fa49.jpeg)

I agree with you about the not immediate readability but it's a good compromise with the several lines of code that I would write using an iteration, besides I guess this is a safe way to catch isbn, because I'm eliminating  all the non string types.

Why would you consider this a  "**very** roundabout" ? What makes it **very** ? 

Thanks for your feedback, unfortunately I'm not finding anything in your comment that helps me to find a different way of writing this code, I'd be happy to talk deeper about this in tomorrow' call :)